src/Nox.Solution/Models/Domain/Entities/Entity.cs

```diff
@@ -25,6 +25,9 @@ public class Entity : DefinitionBase
     [YamlIgnore]
     private ConcurrentDictionary<string, NoxSimpleTypeDefinition>? _keysByName;
 
+    [YamlIgnore]
+    private ConcurrentDictionary<EntityRelationship, string>? _relationshipPublicNamesByRelationship;
```

Comment:
@anna-naumova25  no need to be a ConcurrentDictionary since apply defaults is called only once on the validation phase, there will no be multiple threads calling this method